src/app/core/services/loader.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { Observable, BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoaderService {
+  private _isLoading: BehaviorSubject<boolean> = new BehaviorSubject(false);
+  private isLoading: Observable<boolean> = this._isLoading.asObservable();
+
+  constructor() { }
+
+  showLoader() {
+    this._isLoading.next(true);
```

Comment:
Imagine the case when you submit 3 requests from different parts of your application in one time. You don't know time when they will be resolved/rejected. With the current implementation this first resolved request will hide the spinner, while 2 others may still be in process. Any ideas how to improve it?)